### PR TITLE
More common types deserializers

### PR DESCRIPTION
**Pull request by alenkacz**

### CODE REVIEW

File: src/main/scala/cz/alenkacz/db/postgresscala/DbValue.scala [hunk 1]
```diff
@@ -1,6 +1,43 @@
...
+  def boolOpt: Option[Boolean]
+  def bools: Array[Boolean]
+  def short: Short
+  def shortOpt: Option[Short]
+  def shorts: Array[Short]
+  def inetAddress: InetAddress
+  def inetAddressOpt: Option[InetAddress]
+  def uuid: UUID
+  def uuidOpt: Option[UUID]
+  def dateTime: Instant
```
Comment (augi) [inline src/main/scala/cz/alenkacz/db/postgresscala/DbValue.scala]: I would name it `instant`

File: src/main/scala/cz/alenkacz/db/postgresscala/PostgresAsyncDbValue.scala [hunk 2]
```diff
@@ -1,6 +1,81 @@
...
+
+  override def time: Time = value.asInstanceOf[Time]
+
+  override def bytes: Array[Byte] = value.asInstanceOf[Array[Byte]]
+
+  override def any: Any = value
+
+  override def stringOpt: Option[String] = Option(string)
+
+  override def strings: Array[String] = value.asInstanceOf[Array[String]]
```
Comment (augi) [inline src/main/scala/cz/alenkacz/db/postgresscala/PostgresAsyncDbValue.scala]: Are you sure that retype to `Array[String]` works? Just asking, maybe a test would be nice.

**Review by augi — CHANGES_REQUESTED** [review]
...

**Comment by alenkacz** [comment]
@augi thanks for forcing me to write integration test, it turned out to be really needed

to make it work, I did even some changes in API so the best thing would probably be to re-review the whole PR, thanks!

File: src/test/scala/cz/alenkacz/db/postgresscala/PostgresqlAsyncIntegrationTest.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  def createTestTable() = {
+    Await.result(connection.execute("""CREATE TABLE IF NOT EXISTS abc (
+           id serial PRIMARY KEY,
+           a text NOT NULL UNIQUE,
+           b text[] NOT NULL
+         );"""), Duration(3, TimeUnit.SECONDS))
+  }
+}
+
+case class TestObject(testString: String, testStringArray: Seq[String])
```
Comment (augi) [inline src/test/scala/cz/alenkacz/db/postgresscala/PostgresqlAsyncIntegrationTest.scala:117]: I would prefer to add more field, ideally all the supported types.

**Comment by alenkacz** [comment]
@augi all supported types are tested via integration test. Is it OK now?

File: src/main/scala/cz/alenkacz/db/postgresscala/PostgresAsyncDbValue.scala [hunk 4]
```diff
@@ -35,43 +35,46 @@ class PostgresAsyncDbValue(value: Any) extends DbValue {
...
   }
 
-  override def time: Time = value.asInstanceOf[Time]
+  override def time: Time = {
+    val yodaTime = value.asInstanceOf[org.joda.time.LocalTime]
+    Time.valueOf(LocalTime.of(yodaTime.getHourOfDay, yodaTime.getMinuteOfHour, yodaTime.getSecondOfMinute))
+  }
 
-  override def bytes: Array[Byte] = value.asInstanceOf[Array[Byte]]
+  override def bytes: Seq[Byte] = value.asInstanceOf[Array[Byte]].toList
```
Comment (augi) [inline src/main/scala/cz/alenkacz/db/postgresscala/PostgresAsyncDbValue.scala]: Would be `toSeq` more efficient than `toList`?

File: src/test/scala/cz/alenkacz/db/postgresscala/PostgresqlAsyncIntegrationTest.scala [hunk 5]
```diff
@@ -13,19 +18,107 @@ import org.scalatest.junit.JUnitRunner
...
-    val expected = TestObject("a", Seq("a", "b"))
-    connection.execute("INSERT INTO abc (a, b) VALUES ('a', '{\"a\", \"b\"}')").flatMap(_ => sql"SELECT a,b FROM abc".query(r => TestObject(r(0).string, r(1).strings))).flatMap(actual => {
-      actual.head should be(expected)
+    connection.execute("INSERT INTO abc (s, sa) VALUES ('a', '{\"a\", \"b\"}')").flatMap(_ => sql"SELECT s,sa FROM abc".query(r => (r(0).string, r(1).strings))).flatMap(actual => {
+      actual.head should be(("a", Seq("a", "b")))
+    })
+  }
+
+  it should "return result for basic int query" in {
+    createTestTable()
```
Comment (augi) [inline src/test/scala/cz/alenkacz/db/postgresscala/PostgresqlAsyncIntegrationTest.scala]: I don't like we have to repeat `createTestTable`. Couldn't it be solved somehow?

File: src/main/scala/cz/alenkacz/db/postgresscala/PostgresAsyncDbValue.scala (hunk 4 above)
Comment (alenkacz) [reply src/main/scala/cz/alenkacz/db/postgresscala/PostgresAsyncDbValue.scala]: the reasoning behind toList was that most of the methods return ArrayBuffer which is mutable and I do not want to expose anything mutable

so I just changed it for this byte array

**Comment by alenkacz** [comment]
@augi both fixed...

**Comment by augi** [comment]
👍

**Review by augi — APPROVED** [review]

**Pull request merged**